fastestimator/op/tensorop/augmentation/mixup_batch.py

```diff
@@ -50,30 +50,49 @@ def __init__(self,
                  mode: Union[None, str, Iterable[str]] = 'train',
                  ds_id: Union[None, str, Iterable[str]] = None,
                  alpha: float = 1.0,
-                 shared_beta: bool = True):
+                 shared_beta: bool = False):
         assert alpha > 0, "MixUp alpha value must be greater than zero"
         super().__init__(inputs=inputs, outputs=outputs, mode=mode, ds_id=ds_id)
-        assert len(self.outputs) == len(self.inputs) + 1, "MixUpBatch requires 1 more output than inputs"
         self.alpha = alpha
         self.beta = None
         self.shared_beta = shared_beta
-        self.in_list, self.out_list = True, True
+        #self.in_list, self.out_list = True, True
 
     def build(self, framework: str, device: Optional[torch.device] = None) -> None:
         if framework == 'tf':
             self.beta = tfp.distributions.Beta(self.alpha, self.alpha)
         elif framework == 'torch':
-            self.beta = torch.distributions.beta.Beta(self.alpha, self.alpha)
+            device = torch.device('cuda' if torch.cuda.is_available() else 'cpu')
+            self.beta = torch.distributions.beta.Beta(
+                torch.tensor([self.alpha]).to(device), torch.tensor([self.alpha]).to(device))
         else:
             raise ValueError("unrecognized framework: {}".format(framework))
 
-    def forward(self, data: List[Tensor], state: Dict[str, Any]) -> List[Tensor]:
+    def forward(self, data: List[Tensor], state: Dict[str, Any]) -> Tuple[Tensor, Tensor]:
+        x, y = data
+
         if self.shared_beta:
             lam = self.beta.sample()
+            lam_y = lam
         else:
-            lam = self.beta.sample(sample_shape=(data[0].shape[0], ))
-            shape = [-1] + [1] * (len(data[0].shape) - 1)
+            if tf.is_tensor(x):
+                shp = tf.shape(x)
+            elif isinstance(x, torch.Tensor):
+                shp = x.shape
+            lam = self.beta.sample(sample_shape=(shp[0], ))
+            shape = [-1] + [1] * (len(shp) - 1)
```

Comment:
can we double check whether manual reshaping is actually required? auto broadcasting might handle this for us which would save some performance